libs/types/src/election.ts

```diff
@@ -619,10 +598,10 @@ export function* findElectionDefinitionErrors(
   }
 }
 
-function* findDuplicateIds<Id, T extends { id: Id }>(
+function* findDuplicateIds<T extends { id: unknown }>(
```

Comment:
This is the proper way to do what I wanted before. Previously it inferred `Id` as `unknown` when nothing was explicitly provided. Now it can infer `T` as a concrete type, e.g. `Precinct`, and get the right type for `T['id']`, e.g. `string`.